source/dialogs/combi.c

```diff
@@ -35,6 +35,7 @@
 #include <stdlib.h>
 
 #include "mode-private.h"
+#include "widgets/textbox.h"
```

Comment:
Why is this include included?

Reply:
It defines the `MARKUP` constant.

Reply:
aah, we should fix that, so we don't need to include widget.

Its been a while since I look at this, how this work if a row does not have MARKUP set but contains a character that needs to be escaped.. f.e. `&`

Reply:
```
(process:2032982): Pango-WARNING **: 18:56:41.978: pango_layout_set_markup_with_accel: Error on line 1: Entity did not end with a semicolon; most likely you used an ampersand character without intending to start an entity — escape ampersand as &amp;
```

Reply:
yep it breaks.

Reply:
To be honest, I don’t know how this will interact with entries from the modes that don’t use markup. If I don’t set `MARKUP` here, entries from `drun` are formatted, but entries from `ssh` are not. Maybe we should escape `str` if it originates from a mode that doesn’t use `MARKUP` (so the text may contain characters that need to be escaped), but how to detect this?

Reply:
the same way you set it.

Reply:
Hmm, I think that I know the way…

Reply:
check if it is set in `state`.

Reply:
Okay, I just update the commit.

Reply:
I misunderstood the code a bit, there’s actually no need for extra `entry_state`.